stuffer/s2n_stuffer.h

```diff
@@ -59,6 +59,7 @@ struct s2n_stuffer {
 #define s2n_stuffer_data_available( s )   ((s)->write_cursor - (s)->read_cursor)
 #define s2n_stuffer_space_remaining( s )  ((s)->blob.size - (s)->write_cursor)
 #define s2n_stuffer_is_wiped( s )         ((s)->high_water_mark == 0)
+#define s2n_stuffer_is_empty( s )         ((s)->blob.data == NULL)
```

Comment:
nit: I think this checking to see if the underlying blob is allocated. I feel I could confuse this function with `s2n_stuffer_data_available`("is there any data in the stuffer? use s2n_stuffer_is_empty?")

Reply:
I could try to figure out a better name, but it's replacing "s2n_stuffer_resize_if_empty" so the "empty" terminology is already there :\ Any suggestions for a clearer name?